core/src/main/java/org/jruby/runtime/callsite/CacheEntry.java

```diff
@@ -15,11 +15,7 @@ public CacheEntry(DynamicMethod method, int token) {
     }
 
     public final boolean typeOk(RubyClass incomingType) {
-        return typeOk(this, incomingType);
-    }
-
-    public static final boolean typeOk(CacheEntry entry, RubyClass incomingType) {
```

Comment:
you need to careful when removing methods on places such as these.
might be called from JIT-ed byte-code (so you're not seeing it show up as used)

Reply:
@kares yea, I learnt that today in some other spot already :) Questions (if you have a sec):

1. Can I do anything besides running tests + full-text search to make sure I don't kill such a method accidentally?
2. Am I correct to understand that the `@JIT` annotation should be put on methods which could be called dynamically like you described?

Thanks!

Reply:
>    Can I do anything besides running tests + full-text search to make sure I don't kill such a method accidentally?

usually just grep for the method-name string -> `"typeOf"` should be safe to find whether its used :)

>    Am I correct to understand that the @JIT annotation should be put on methods which could be called dynamically like you described?

well yeah - have run into that as well ... lately.
but what I dislike is that now some places might have a dependency on the IR package (`org.jruby.ir.JIT`) -> should maybe get relocated somewhere else I guess ... not sure
haven't raised that concern yet -> you should chat with @enebo or @headius on IRC if it was meant to be somehow IR specific or not.

Reply:
@kares thanks!